apps/prairielearn/src/pages/administratorCourses/administratorCourses.html.ts

```diff
@@ -0,0 +1,384 @@
+                configuration file in the repository during the sync.
+              </small>
+            </div>
+          </div>
+        </main>
+      </body>
+    </html>
+  `.toString();
+}
+
+function CourseDeleteForm({
+  id,
+  course,
+  csrfToken,
+}: {
+  id: string;
+  course: CourseWithInstitution;
+  csrfToken: string;
+}) {
+  return html`
+    <form name="add-user-form" method="POST">
+      <input type="hidden" name="__action" value="courses_delete" />
+      <input type="hidden" name="__csrf_token" value="${csrfToken}" />
+      <input type="hidden" name="course_id" value="${course.id}" />
+      <div class="form-group">
+        <label for="inputConfirm${id}">Type "${course.short_name}" to confirm:</label>
+        <input type="text" class="form-control" id="inputConfirm${id}" name="confirm_short_name" />
+      </div>
+      <div class="text-right">
+        <button type="button" class="btn btn-secondary" onclick="$('#${id}').popover('hide')">
+          Cancel
+        </button>
+        <button type="submit" class="btn btn-danger">Delete course</button>
+      </div>
+    </form>
+  `;
+}
+
+function CourseInsertForm({
+  id,
+  institutions,
+  csrfToken,
+}: {
+  id: string;
+  institutions: Institution[];
+  csrfToken: string;
+}) {
+  return html`
+    <form name="add-course-form" method="POST">
+      <input type="hidden" name="__action" value="courses_insert" />
+      <input type="hidden" name="__csrf_token" value="${csrfToken}" />
+      <div class="form-group">
+        <label>Institution:</label>
+        <select
+          name="institution_id"
+          class="form-control"
+          onchange="this.closest('form').querySelector('[name=display_timezone]').value = this.querySelector('option:checked').dataset.timezone;"
+        >
+          ${institutions.map((i) => {
+            return html`
+              <option value="${i.id}" data-timezone="${i.display_timezone}">${i.short_name}</option>
+            `;
+          })}
+        </select>
+      </div>
+      <div class="form-group">
+        <label for="courseAddInputShortName">Short name:</label>
+        <input
+          type="text"
+          class="form-control"
+          id="courseAddInputShortName"
+          name="short_name"
+          placeholder="XC 101"
+        />
+      </div>
+      <div class="form-group">
+        <label for="courseAddInputTitle">Title:</label>
+        <input
+          type="text"
+          class="form-control"
+          id="courseAddInputTitle"
+          name="title"
+          placeholder="Template course title"
+        />
+      </div>
+      <div class="form-group">
+        <label for="courseAddInputTimezone">Timezone:</label>
+        <input
+          type="text"
+          class="form-control"
+          id="courseAddInputTimezone"
+          name="display_timezone"
+          value="${institutions[0]?.display_timezone}"
+        />
```

Comment:
Do we want to consider converting this to a dropdown?

Reply:
https://en.wikipedia.org/wiki/List_of_tz_database_time_zones?wprov=sfla1

The list is big...

Reply:
@jonatanschroeder we have prior art for this:

https://github.com/PrairieLearn/PrairieLearn/blob/479268edd99e96dbc4f47659333f0774c72ccef7/apps/prairielearn/src/pages/administratorInstitutions/administratorInstitutions.html.ts#L95

@mylesw27 there's just enough complexity associated with this (specifically updating the selected timezone based on the selected institution) that I'll hold off on that for now. This form is also essentially never used in practice, so it shouldn't be too much of a usability issue.